root/programs/C#/Frameworks/Infrastructure/ServiceInterface/ASPNETWebService/ASPNETWebService/Controllers/FxController.cs

```diff
@@ -0,0 +1,343 @@
+//* 作成日時        ：－
+//* 作成者          ：生技
+//* 更新履歴        ：
+//* 
+//*  日時        更新者            内容
+//*  ----------  ----------------  -------------------------------------------------
+//*  2017/08/18  西野 大介         新規作成
+//**********************************************************************************
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Web;
+using System.Web.Http;
+using System.Web.Http.Cors;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+using Touryo.Infrastructure.Business.Presentation;
+using Touryo.Infrastructure.Framework.Exceptions;
+using Touryo.Infrastructure.Framework.Common;
+using Touryo.Infrastructure.Framework.Util;
+using Touryo.Infrastructure.Framework.Transmission;
+
+using Touryo.Infrastructure.Public.Db;
+using Touryo.Infrastructure.Public.IO;
+using Touryo.Infrastructure.Public.Log;
+using Touryo.Infrastructure.Public.Str;
+using Touryo.Infrastructure.Public.Util;
+
+namespace ASPNETWebService.Controllers
+{
+    /// <summary>
+    /// ASP.NET WebAPI JSON-RPCの.NETオブジェクトの
+    /// バイナリ転送用メソッドを公開するサービス インターフェイス基盤
+    /// </summary>
+    [EnableCors(
+        // リソースへのアクセスを許可されている発生元
+        origins: "*",
+        // リソースによってサポートされているヘッダー
+        headers: "*",
+        // リソースによってサポートされているメソッド
+        methods: "*",
+        // 
+        SupportsCredentials = true)]
+    public class FxController : ApiController
+    {
+        #region テスト用
+
+        /// <summary>test</summary>
+        /// <returns>Dictionary(string, string)</returns>
+        [HttpGet]
+        [Route("test")]
+        public Dictionary<string, string> test()
+        {
+            Dictionary<string, string> dic = new Dictionary<string, string>();
+            dic.Add("AAA", "aaa");
+            dic.Add("BBB", "bbb");
+            dic.Add("CCC", "ccc");
+
+            return dic;
+        }
+
+        #endregion
+
+        #region グローバル変数
+
+        /// <summary>インプロセス呼び出しの名前解決シングルトン クラス</summary>
+        /// <remarks>
+        /// 初期化は起動時の１回のみであり、
+        /// 読み取り専用のデータを保持する場合
+        /// のみに適用するデザインパターンとする。
+        /// </remarks>
+        private static InProcessNameService IPR_NS = new InProcessNameService();
+
+        #endregion
+
+        #region ASP.NET WebAPI JSON-RPCの.NETオブジェクトのバイナリ転送用メソッド
+
+        /// <summary>
+        /// ASP.NET WebAPI JSON-RPCの.NETオブジェクトのバイナリ転送用メソッド
+        /// </summary>
+        /// <param name="paramDic">
+        /// 引数：Dictionary(string, string)
+        /// ・ServiceName
+        /// ・ContextObject
+        /// ・ParameterValueObject
+        /// </param>
+        /// <returns>
+        /// 戻り値：Dictionary(string, string)
+        /// ・Return
+        /// ・ContextObject
+        /// ・ReturnValueObject
+        /// </returns>
+        [HttpPost]
+        [Route("DotNETOnlineWAPI")]
+        public Dictionary<string, string> DotNETOnlineWAPI(Dictionary<string, string> paramDic)
```

Comment:
@daisukenishino 

The developers might not understand the meaning of *WAPI*.
It might be a good idea to rename this method to *DotNETOnlineWebAPI*.